backend/de.metas.manufacturing.rest-api/src/main/java/de/metas/manufacturing/job/service/ManufacturingJobService.java

```diff
@@ -513,25 +518,32 @@ public ManufacturingJob receiveGoods(
 			@NonNull final FinishedGoodsReceiveLineId lineId,
 			@NonNull final SelectedReceivingTarget receivingTarget,
 			@NonNull final BigDecimal qtyToReceiveBD,
-			@NonNull final ZonedDateTime date)
-	{
-		final ManufacturingJob changedJob = job.withChangedReceiveLine(lineId, line -> {
-			final ReceivingTarget newReceivingTarget = trxManager.callInThreadInheritedTrx(() -> ReceiveGoodsCommand.builder()
-					.handlingUnitsBL(handlingUnitsBL)
-					.ppOrderBL(ppOrderBL)
-					.ppOrderBOMBL(ppOrderBOMBL)
-					.loadingAndSavingSupportServices(loadingAndSavingSupportServices)
-					//
-					.ppOrderId(job.getPpOrderId())
-					.coProductBOMLineId(line.getCoProductBOMLineId())
-					.receivingTarget(receivingTarget)
-					.qtyToReceiveBD(qtyToReceiveBD)
-					.date(date)
-					//
-					.build().execute());
-
-			return line.withReceivingTarget(newReceivingTarget);
-		});
+			@NonNull final ZonedDateTime date,
```

Comment:
I think the number of parameters is growing a lot here. Consider introducing ReceiveGoodsRequest